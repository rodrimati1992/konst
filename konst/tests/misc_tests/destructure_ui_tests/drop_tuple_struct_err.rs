struct Tuple(String, String, String);

impl Drop for Tuple {
    fn drop(&mut self) {}
}

const fn to_bar(foo: Tuple) -> (String, String, String) {
    konst::destructure!{Tuple(foo, bar, baz) = foo}

    (foo, bar, baz)
}

fn main(){}