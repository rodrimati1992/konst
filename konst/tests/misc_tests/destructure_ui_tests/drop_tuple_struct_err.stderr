error[E0308]: mismatched types
 --> tests/misc_tests/destructure_ui_tests/drop_tuple_struct_err.rs:8:5
  |
8 |     konst::destructure!{Tuple(foo, bar, baz) = foo}
  |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `__DoesNotImplDrop<Tuple>`, found `__ImplsDrop<Tuple>`
  |
  = note: expected struct `__DoesNotImplDrop<Tuple>`
             found struct `__ImplsDrop<Tuple>`
  = note: this error originates in the macro `$crate::__destructure_struct` which comes from the expansion of the macro `konst::destructure` (in Nightly builds, run with -Z macro-backtrace for more info)
